src/components/profiles-modals/ImportProfileModal.vue

```diff
@@ -117,9 +117,17 @@ export default class ImportProfileModal extends mixins(ProfilesMixin) {
             return;
         }
 
-        let read: string | null = await ProfileUtils.readProfileFile(files[0]);
+        let read: string = '';
+        try {
+            read = await ProfileUtils.readProfileFile(files[0]);
+        } catch (e: unknown) {
+            const err = R2Error.fromThrownValue(e);
+            this.$store.commit('error/handleError', err);
+            this.closeModal();
+            return;
+        }
 
-        if (read !== null) {
+        if (read) {
```

Comment:
I think this `if`-check might not be needed, as the previous "step" should have returned if `read` is falsy? And if it somehow is false, we silently skip setting two important looking state variables, which might make any bugs harder to track? Unless you've got counter arguments, I'd prefer it removed.